### PR TITLE
Tic-Tac-Toe Milestone 3: Add logic to the game

### PR DESCRIPTION
**Pull request by sergioobolevich**
In this project we are creating Tic-Tac-Toe game. 
We have added the logic (classes and methods) to the game in this milestone.
We have created 2 classes: Board and Player with some methods on them.
Player just saves player name and team. And Board does the most.
Board class has the methods:  make_move, check_if_win?, check_if_move_done? and show_board .
This classes are used on main.rb to execute the game. I hope you like it.

### CODE REVIEW

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUESTED 🛑 

Hello team!
Great job so far with this project! 👏 
Descriptive commits and good readme file! 💯 

However, we need to implement some changes before this project gets approved 💪 
Kindly check my comments below 👇 apply the changes and resubmit your project for a new code review ⏩ 

## Mandatory changes
- In your readme file the instructions are not right, users should run the command 'ruby ./bin/main.rb', kindly modify it.
- You have used 2 classes (board, player) but looking at your project maybe you need a third one(game) where you can implement all the logic of the game, now many of those methods have no class or are in the wrong class(for example, chef_if_win? method). Please implement a new class with all the game logic. **Hint:** Use your main.rb to ask the users for the data and initialize the game there, calling the game class, and in your lib folder implement all classes and logic
Some bugs I have found in the project: 
   - The application should not allow a player to move into an occupied position, kindly fix it
   - The application should save players move and display the status of the board after that, now the board always show the numbers from 1 to 9, when it should appear an X or a 0. Kindly, fix that. You can add a update_board method in your board class.
- It would be nice to display the board before the first move, so the players know where are they moving.
- When players select play again, should not re-enter their names again, maybe the team but not the name. Make it happen

**Comment by sergioobolevich** [comment]
Hello @javitocor , sorry I ask you to recheck or clarify us this:

>   * The application should not allow a player to move into an occupied position, kindly fix it

Game it's not currently allowing to move into an occupied position. It asks you to make the move again when the position has already been taken.

>   * The application should save players move and display the status of the board after that, now the board always show the numbers from 1 to 9, when it should appear an X or a 0. Kindly, fix that. You can add a update_board method in your board class.

The application is changing the number with the X and O on each move and displaying the modified board..

> * It would be nice to display the board before the first move, so the players know where are they moving.

The board is being displayed before the first move..

Maybe we didn't understand what you meant, or you confused the project. Thanks for clarifying!

File: lib/logic.rb
Comment (kevcap) [inline lib/logic.rb:51]: ⚠ [OPTIONAL CHANGE] ⚠
- [ ] There should be no puts outisde of the main.rb file.

**Review by kevcap — APPROVED** [review]
# [STATUS] PROJECT APPROVED 🍾

### Hello @chubaquelo and @deikdesign I tested your project playing TicTacToe with some friends and everything is working fine 😁. So you have my approval to go ahead and merge this pull request! Despite that, I left some optional changes.

[OPTIONAL CHANGES]
- [ ] The project looks nice, but I recommend you guys to use only `| ` and `_` or `-` without `+`. I get a bit confused.
**⚠These are mandatory changes, but since It wasn't pointed out earlier, I'll mark them as [OPTIONAL]:**
- [ ] There should be no puts outside of the main.rb file, and you guys have it at the logic.rb on the show_board method.
That's all boys, congrats again 🥇

### It was a pleasure to review your project!


## As always, I wish you all the very best and see you both at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Pull request merged**
